src/hepattn/models/decoder.py

```diff
@@ -250,10 +252,14 @@ def __init__(
             bidirectional_ca: Enable bidirectional cross-attention.
             qkv_norm: Apply normalization to QKV in attention.
             hybrid_norm: Enable hybrid normalization from 2503.04598.
+            legacy: Use legacy attention implementation.
+            skip_first_layer_pe: Skip positional encoding for the first layer.
         """
         super().__init__()
         self.dim = dim
         self.bidirectional_ca = bidirectional_ca
+        self.skip_first_layer_pe = skip_first_layer_pe
+        self.legacy = legacy
```

Comment:
let's remove both of these and use the non-legacy with skip_first_layer_pe=False